packages/std/src/mock.rs

```diff
@@ -126,22 +127,38 @@ pub fn mock_env<T: Api, U: Into<HumanAddr>>(api: &T, sender: U, sent: &[Coin]) -
     }
 }
 
+/// The same type as cosmwasm-std's QuerierResult, but easier to reuse in
+/// cosmwasm-vm. It might diverge from QuerierResult at some point.
+pub type MockQuerierCustomHandlerResult = SystemResult<StdResult<Binary>>;
+
 /// MockQuerier holds an immutable table of bank balances
 /// TODO: also allow querying contracts
-#[derive(Clone, Default)]
-pub struct MockQuerier {
+pub struct MockQuerier<C: DeserializeOwned = Never> {
     bank: BankQuerier,
     staking: StakingQuerier,
     // placeholder to add support later
     wasm: NoWasmQuerier,
+    /// A handler to handle custom queries. This is set to a dummy handler that
+    /// always errors by default. Update it via `with_custom_handler`.
+    ///
+    /// Use box to avoid the need of another generic type
+    custom_handler: Box<dyn for<'a> Fn(&'a C) -> MockQuerierCustomHandlerResult>,
 }
 
-impl MockQuerier {
+impl<C: DeserializeOwned> MockQuerier<C> {
     pub fn new(balances: &[(&HumanAddr, &[Coin])]) -> Self {
+        // strange argument notation suggested as a workaround here: https://github.com/rust-lang/rust/issues/41078#issuecomment-294296365
+        let no_handler = |_: &_| -> MockQuerierCustomHandlerResult {
```

Comment:
Good one. Weird, but if it works...
We never need to touch this as devs, but interesting placeholder notation.